leapp/utils/report.py

```diff
@@ -42,31 +39,23 @@ def fetch_upgrade_report_raw(context_id, renderers=True):
             'flags': data['flags'],
             'severity': data['severity']
         }
-        if renderers:
-            merged_data['renderers'] = data['renderers']
         detail = _flatten_dict(json.loads(data['detail']))
         merged_data.update(detail)
         messages.append(merged_data)
 
     return messages
 
 
-def fetch_upgrade_report_messages(context_id, renderer):
-    """
-    :param context_id: ID to identify the needed messages
-    :type context_id: str
-    :param renderer: How messages will be rendered (e.g html or plaintext)
-    :type renderer: str
-    :return: All upgrade messages of type "Report" withing the given context
-    """
-    messages_raw = fetch_upgrade_report_raw(context_id)
-    if not messages_raw:
-        # return [] to be consistent?
-        return
-
-    messages_to_print = []
-    for msg_data in messages_raw:
-        template = Template(msg_data.pop('renderers').get(renderer))
-        messages_to_print.append(template.render(msg_data))
-
-    return messages_to_print
+def generate_report_file(messages_to_report, path):
+    if path.endswith(".txt"):
+        with open(path, 'w') as f:
+            for message in messages_to_report:
+                f.write('Severity: {}\n'.format(message['severity']))
```

Comment:
I am wondering what about flags, audiences, tags, external links. Some of those does not need to be in the plaintext report, e.g. tags, but links should definitely be there.

Question is, how to best render those. Currently it seems that:

`f.write('Remediation: {}\n'.format(message['detail']['remediation']))`

will be more JSON-like as ['remediation'] will return list of remediations and remediations may contain another list of commands.

We discussed this briefly with @shaded-enmity, we probably does not need to take care of formatting commands but should avoid some deep structures in our plaintext report. 

But it is matter of discussion of course.

Reply:
Sure, I'll add anything we consider necessary, for now as the initial commit I intended to keep the basic functionality as it is.